Import users from covidhub

ttung opened the pull request:
### Description
Interface with auth0 and covid database to build up users.  In addition to the group associated with the RR project, we also import users associated with CDPH and admin.

Depends on #172, #174, https://github.com/czbiohub/covidhub/pull/1512, https://github.com/czbiohub/covidhub/pull/1513

#### Issue
[ch125195](https://app.clubhouse.io/genepi/stories/space/125195)

### Test plan
imported users successfully and verified manually on db.

phoenixAja reviewed (changes requested):
looks good, but do you also need to add the piplock file if the requirement dependencies have changed?

ttung commented:
Update `Pipfile` and `Pipfile.lock`

phoenixAja commented on src/py/aspen/covidhub_import/implementation.py:
```diff
@@ -48,6 +55,13 @@
 
+@dataclass
+class Auth0Entry:
+    nickname: str
```
we don't need to add username to this? or does nickname mean username?

ttung replied:
For the covidtracker auth0 account, the usernames were all email addresses.

phoenixAja replied:
oh ok then nevermind

phoenixAja commented on src/py/aspen/covidhub_import/implementation.py:
```diff
@@ -97,24 +126,85 @@ def import_project(
+                project.originating_lab,
+                project.originating_address,
+                # FIXME: this needs to be updated.
+                email="fakeemail@dph.gov",
```
i thought that this field had to be unique, or did we disable that until we have a workaround?

phoenixAja replied:
will this break if we are moving over multiple groups (besides admin and cdph)?

ttung replied:
Yeah, this will break with multiple groups.

phoenixAja commented on src/py/aspen/covidhub_import/implementation.py:
```diff
@@ -97,24 +126,85 @@ def import_project(
+        covidhub_group_to_aspen_group = {
+            project.covidtracker_group.group: group,
+            covidhub_session.query(covidtracker.Group)
+            .filter(covidtracker.Group.name == "Admin")
```
does this need to be `"ADMIN"`?

ttung replied:
No, in the old DB, it was "Admin".

phoenixAja commented on src/py/aspen/covidhub_import/implementation.py:
```diff
@@ -97,24 +126,85 @@ def import_project(
+            or_(
+                covidtracker.UsersGroups.group_id.in_(
+                    covidhub_session.query(covidtracker.Group.id).filter(
+                        covidtracker.Group.name.in_(["Admin", "CDPH"])
```
same here? Or is this not case sensitive?

ttung replied:
Same here, in the old DB, it was Admin and CDPH.

phoenixAja commented on src/py/aspen/covidhub_import/implementation.py:
```diff
@@ -97,24 +126,85 @@ def import_project(
+            if user is None:
+                user = User()
+
+            user.name = auth0_user.nickname
```
this is a nitpick, but i feel like we should rename `nickname` to `name`.

ttung replied:
We're looking at the auth0 nickname field, so it makes sense.  Had we populated our auth0 db correctly, we would be looking at username and not nickname.

phoenixAja replied:
makes sense, keep nickname then 👍

phoenixAja reviewed (commented):
looks good for the most part just had some comments

Commit:
Import users from covidhub

### Description
Interface with auth0 and covid database to build up users.  In addition to the group associated with the RR project, we also import users associated with CDPH and admin.

#### Issue
[ch125195](https://app.clubhouse.io/genepi/stories/space/125195)

### Test plan
imported users successfully and verified manually on db.
